File: LeetCode/Medium/MaximumSubarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int ans = nums[0];
        int prev_max = ans;

        for (int i = 1; i < nums.size(); i++) {
            prev_max = max(nums[i], prev_max + nums[i]);
            ans = max(prev_max, ans);
        }

        return ans;
    }
};

File: LeetCode/Medium/MaximumSubarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        prev_max = nums[0]
        ans = prev_max
        N = len(nums)

        for i in range(1, N):
            prev_max = max(prev_max + nums[i], nums[i])
            ans = max(prev_max, ans)

        return ans